### DISCUSSION TITLE
Point on stereo-camera that corresponds with camera-pose

### DISCUSSION THREAD

**Discussion by robin-zealrobotics** [stella-cv/stella_vslam#533] [Q&A]
When using stella with a stereo-camera (e.g. Realsense D455), what physical point on the camera does camera-pose represent?
Is it the left/right lens or the middle of the two?
And as far as depth-wise, is it the location of the image sensor, the lens, some focal-length-dependent thing, ...?

In essence, I'm trying to figure out the correct way to have a transform from base_link to camera_link with regards to stella. And for that I need to know how camera_link (for stella) is 'attached' to my stereo camera.

**Comment by jjd9** [comment]
(since you are using ROS TF terminology, e.g. base_link, I will assume you are using stella_vslam_ros)

Please have a look at this illustration, I will use it for more clarity: https://github.com/IntelRealSense/realsense-ros?tab=readme-ov-file#ros2robot-vs-opticalcamera-coordination-systems

In stereo mode, the SLAM algorithm tracks the pose of the left camera in what the illustration i linked calls the "optical" coordinate system 

realsense_ros calls this the `camera_infra1_optical_frame`. (infra1 is the left IR camera)

However, the node converts this pose to the ROS2(Robot) coordinate system before publishing it. https://github.com/stella-cv/stella_vslam_ros/blob/d75891885e3c4f8745f2cd9bfc5f94f32090ac29/src/stella_vslam_ros.cc#L64

realsense_ros calls this the `camera_infra1_frame`

if you look at your TF tree, you should see that your robot is connected to the realsense via a link called the `camera_bottom_screw_frame` and if you follow the TF tree, you will see the `camera_infra1_frame`
assuming your setup is correct, you should be able to transform the pose stella_vslam_ros gives you in the `camera_infra1_frame` to the base_link frame via TF.

**Reply by jjd9** [reply]
note that this assumes you have setup the `..._frame` parameters in stella_vslam_ros correctly. otherwise the pose message may not be stamped properly.
